v2/blockstore/readwrite_test.go

```diff
@@ -688,3 +687,165 @@ func TestReadWriteErrorAfterClose(t *testing.T) {
+	require.NoError(t, err)
+	err = subject.Put(idBlock)
+	require.NoError(t, err)
+
+	has, err := subject.Has(idCid)
+	require.NoError(t, err)
+	require.True(t, has)
+
+	gotBlock, err := subject.Get(idCid)
+	require.NoError(t, err)
+	require.Equal(t, idBlock, gotBlock)
+
+	keysChan, err := subject.AllKeysChan(context.Background())
+	require.NoError(t, err)
+	var i int
+	for c := range keysChan {
+		i++
+		require.Equal(t, idCid, c)
+	}
+	require.Equal(t, 1, i)
+
+	err = subject.Finalize()
+	require.NoError(t, err)
+
+	// Assert resulting CAR file indeed has the IDENTITY block.
+	f, err := os.Open(path)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, f.Close()) })
+
+	reader, err := carv2.NewBlockReader(f)
+	require.NoError(t, err)
+
+	gotBlock, err = reader.Next()
+	require.NoError(t, err)
+	require.Equal(t, idBlock, gotBlock)
+
+	next, err := reader.Next()
+	require.Equal(t, io.EOF, err)
+	require.Nil(t, next)
+
+	// Assert the id is indexed.
+	r, err := carv2.OpenReader(path)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, r.Close()) })
+	require.True(t, r.Header.HasIndex())
+
+	ir := r.IndexReader()
+	require.NotNil(t, ir)
+
+	gotIdx, err := index.ReadFrom(ir)
+	require.NoError(t, err)
+
+	// Determine expected offset as the length of header plus one
+	header, err := carv1.ReadHeader(r.DataReader())
+	require.NoError(t, err)
+	object, err := cbor.DumpObject(header)
+	require.NoError(t, err)
+	expectedOffset := len(object) + 1
+
+	// Assert index is iterable and has exactly one record with expected multihash and offset.
+	switch idx := gotIdx.(type) {
+	case index.IterableIndex:
+		var i int
+		err := idx.ForEach(func(mh multihash.Multihash, offset uint64) error {
+			i++
+			require.Equal(t, idmh, mh)
+			require.Equal(t, uint64(expectedOffset), offset)
+			return nil
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, i)
+	default:
+		require.Failf(t, "unexpected index type", "wanted %v but got %v", multicodec.CarMultihashIndexSorted, idx.Codec())
+	}
+}
+
+func TestOpenReadWrite_ErrorsWhenWritingTooLargeOfACid(t *testing.T) {
+	maxAllowedCidSize := uint64(2)
+	path := filepath.Join(t.TempDir(), "readwrite-with-id-enabled-too-large.car")
+	subject, err := blockstore.OpenReadWrite(path, []cid.Cid{}, carv2.MaxIndexCidSize(maxAllowedCidSize))
+	t.Cleanup(subject.Discard)
+	require.NoError(t, err)
+
+	data := []byte("monsterlobster")
+	mh, err := multihash.Sum(data, multihash.SHA2_256, -1)
+	require.NoError(t, err)
+	bigCid := cid.NewCidV1(uint64(multicodec.Raw), mh)
+	bigCidLen := uint64(bigCid.ByteLen())
+	require.True(t, bigCidLen > maxAllowedCidSize)
+
+	bigBlock, err := blocks.NewBlockWithCid(data, bigCid)
+	require.NoError(t, err)
+	err = subject.Put(bigBlock)
+	require.Equal(t, carv2.NewErrCidTooLarge(maxAllowedCidSize, bigCidLen), err)
+}
+
+func TestReadWrite_ReWritingCARv1WithIdentityCidIsIdenticalToOriginalWithOptionsEnabled(t *testing.T) {
+	originalCARv1Path := "../testdata/sample-v1.car"
```

Comment:
so `sample-v1.car` contains identity CIDs? is there already coverage of the case where this fixture gets written _without_ identity CIDs?

Reply:
> so `sample-v1.car` contains identity CIDs? 

Yes it does.

> is there already coverage of the case where this fixture gets written _without_ identity CIDs?

Not sure; great shout. Adding. Thanks 🍻

Reply:
Looks like there is yes. The test [here](https://github.com/ipld/go-car/blob/9f9f687b3606dae462007d93cc0e8684129435f5/v2/blockstore/readwrite_test.go#L50-L126) asserts that when the option is not set the resulting CARv2, written using `ReadWrite` blockstore does not include `IDENTITY` CIDs despite `Put`ing them.

Please shout if I have missed anything.